src/HttpClient.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace Jobcloud\KafkaSchemaRegistryClient;
+
+use Jobcloud\KafkaSchemaRegistryClient\Interfaces\ErrorHandlerInterface;
+use Jobcloud\KafkaSchemaRegistryClient\Interfaces\HttpClientInterface;
+use Psr\Http\Client\ClientExceptionInterface;
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class HttpClient implements HttpClientInterface
+{
+    /**
+     * @var ClientInterface
+     */
+    private $client;
+
+    /**
+     * @var string
+     */
+    private $baseUrl;
+
+    /**
+     * @var string|null
+     */
+    private $username;
+
+    /**
+     * @var string|null
+     */
+    private $password;
+
+    /**
+     * @var RequestFactoryInterface
+     */
+    private $requestFactory;
+
+    /**
+     * @var ErrorHandlerInterface
+     */
+    private $errorHandler;
+
+    /**
+     * @param ClientInterface $client
+     * @param RequestFactoryInterface $requestFactory
+     * @param ErrorHandlerInterface $errorHandler
+     * @param string $baseUrl
+     * @param string|null $username
+     * @param string|null $password
+     */
+    public function __construct(
+        ClientInterface $client,
+        RequestFactoryInterface $requestFactory,
+        ErrorHandlerInterface $errorHandler,
+        string $baseUrl,
+        ?string $username = null,
+        ?string $password = null
+    ) {
+        $this->client = $client;
+        $this->baseUrl = trim($baseUrl, '/');
```

Comment:
```suggestion
        $this->baseUrl = $baseUrl;
```

Reply:
let's leave the trims for now, we can always add them, if we see this causes people problems